codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/parse/JsonParserGenerator.kt

```diff
@@ -205,7 +203,10 @@ class JsonParserGenerator(
 
     private fun RustWriter.deserializeStructInner(members: Collection<MemberShape>) {
         if (members.isEmpty()) {
+            rustTemplate("#{skip_value}(tokens)?;", *codegenScope)
             return
+        } else {
+            rustTemplate("#{expect_start_object}(tokens.next())?;", *codegenScope)
         }
         objectKeyLoop {
```

Comment:
if we actually put the members.isEmpty check inside of `objectKeyLoop`, I think we can remove the special casing and the other changes around consuming vs. not consuming the start element? It also makes it a little clearer and more consistent IMO.

Am I correct in saying that if we fully removed the members.isEmpty() branch the code would be correct (but maybe it would fail a clippy check?)

Reply:
> if we actually put the members.isEmpty check inside of objectKeyLoop, I think we can remove the special casing and the other changes around consuming vs. not consuming the start element? It also makes it a little clearer and more consistent IMO.

I think overall, it's simpler, but it does require the introduction of a `skip_to_end` variation of `skip_value` that assumes the start object/array has already been consumed.

> Am I correct in saying that if we fully removed the members.isEmpty() branch the code would be correct (but maybe it would fail a clippy check?)

Yeah, that's correct.